astropy/time/formats.py

```diff
@@ -370,15 +366,22 @@ def set_jds(self, val1, val2):
         self.jd1 = tm._time.jd1
         self.jd2 = tm._time.jd2
 
-    @property
-    def value(self):
-        # when we get here, getattr will already have ensured our scale
-        # equals epoch scale, so we can just subtract the epoch and convert
-        time_from_epoch = ((self.jd1 - self.epoch.jd1) +
-                           (self.jd2 - self.epoch.jd2)) / self.unit
+    def value(self, parent=None):
+        # Make sure that scale is the same as epoch scale so we can just
+        # subtract the epoch and convert
+        if self.scale != self.epoch_scale:
+            tm = parent.replicate()
```

Comment:
In other words, like @mhvk suggested have some kind of `TimeFormatter._to_value()` method that serves the purpose that's been tacked on to `TimeFormatter.value` here.  The one difference is I would keep it an internal method.